### DISCUSSION TITLE
Default value for missing data in Dynatrace scaler

### DISCUSSION THREAD

**Discussion by zip-chanko** [kedacore/keda#6054] [Q&A / Need Help]
Is there a way to set default value when the metric is returning empty value?

Example Dynatrace query response body.
```
{
  "totalCount": 0,
  "nextPageKey": null,
  "resolution": "10m",
  "result": [
    {
      "metricId": "MyCustomEvent:filter(eq("someProperty","someValue")):count:splitBy("dt.entity.process_group"):fold",
      "dataPointCountRatio": 0,
      "dimensionCountRatio": 0,
      "data": []
    }
  ]
}
```
---
Example `Warning` message from `Scaledobjects`.
```
0s (x11 over 5m1s)     Warning   KEDAScalerFailed          ScaledObject/dynatrace-scaled-object   dynatrace response does not contain any metric series
```

**Comment by JorTurFer** [comment]
Hello
Currently it's not supported. Can't it be set durectly within the query? (I have no idea tbh).
If setting the default value as part of the query isn't allowed, are you willing to open a PR adding the support to the scaler?

**Comment by rickbrouwer** [comment]
The [default transformation](https://docs.dynatrace.com/docs/dynatrace-api/environment-api/metric-v2/metric-selector#default) replaces null values in the payload with the specified value. Is this something you can use?
